### DISCUSSION TITLE
Having problems with template errors?

### DISCUSSION THREAD

**Discussion by aderusha** [HASwitchPlate/HASPone#93] [General]
Over the last week several people have been reporting template errors and I'm struggling to piece together just what the heck is happening.  On my own test systems this has never happened and the traces I've received have me scratching my head.

It's obviously an issue, and I want to get to the bottom of it!  I'm going to be [hanging out on Discord](https://discord.gg/FvDu8SXSvJ) today, if you are experiencing this problem and have some time to work through it with me, maybe [jump on to Discord](https://discord.gg/FvDu8SXSvJ) and give me a shout.  It'd really help me out!

**Comment by aderusha** [comment]
_edit: ignore all this, actual fix is a couple replies below_

One suggestion that might help clear things out in your installation:

1. Configuration > Automations > HASP Core functionality and click the Edit button (pencil icon)
2. Top right ⋮ menu > Delete Automation
3. Configuration > Devices > <Your HASP device name> > DELETE
4. Repeat step 3 for any additional devices with the same name
5. Restart the HASPone device
6. Re-deploy the HASP Core functionality, select the HASPone device, SAVE, RUN ACTIONS

**Comment by deltaroe** [comment]
Throwing some more observations into the mix here. It appears that a freshly installed unit doesn't have any rgb_color values in the *selected_*ground_color entites. For example here's a screenshot of my selected_foreground_color state 
![image](https://user-images.githubusercontent.com/5428503/116827688-3717ba00-ab4f-11eb-8d2a-473bc7b9ad3e.png)

It looks like the previous steps to the one failing seem to be attempting to set a default so no idea why that's not working. The theme blueprints seem to be setting the missing values which seems to be why the solution found here works. https://github.com/HASwitchPlate/HASPone/discussions/89

Maybe set the default rgb_color values when instantiating the color object in case they aren't found in the entity? https://github.com/HASwitchPlate/HASPone/blob/main/Home_Assistant/blueprints/hasp_Core_Functionality.yaml#L450

**Reply by deltaroe** [reply]
May have been a bit premature on calling this solved. I think alot of the issue was I renamed the device in HA and a lot of the blueprint templates seem to rely on that name being consistent when building mqtt topics

**Comment by aderusha** [comment]
If anyone can take a minute to test this out it would help a lot!

I've addressed the issue above through a combination of firmware and blueprint changes, pushing the theme RGB light discovery process into the firmware so the blueprint doesn't need to rely on creating and then using the RGB light entities in the same process.

_edit: test looks good, moved 1.02 to release!_

## HASPone v 1.02 upgrade procedure

1. HASPone web admin > Update Firmware > Update ESP8266 from URL, then click "Update ESP from URL" and allow the firmware update to complete
2. In Home Assistant, go to Configuration > Blueprints and hit the trash can icon next to "HASP Core functionality" to delete the existing copy.
3. Click this here button to deploy the 1.02 version of the Core Functionality blueprint:
[![Open your Home Assistant instance and show the blueprint import dialog with a specific blueprint pre-filled.](https://my.home-assistant.io/badges/blueprint_import.svg)](https://my.home-assistant.io/redirect/blueprint_import/?blueprint_url=https%3A%2F%2Fraw.githubusercontent.com%2FHASwitchPlate%2FHASPone%2Fdev-1.02%2FHome_Assistant%2Fblueprints%2Fhasp_Core_Functionality.yaml)
4. Finally, go to Configuration > Server Controls > YAML configuration reloading > Automations to reload the core functionality blueprints+automations.

**Reply by deltaroe** [reply]
As long as I don't rename the device, it works great. Thanks!

**Reply by aderusha** [reply]
Renaming the device should create a new device and new entities associated with it.  You'll then need to update all of your deployed blueprints to point at the new device.

**Reply by Is0mniac** [reply]
You're a wizard, it works!! Thank you, appreciate the fix.

**Reply by MihleTX** [reply]
This finally solved my issue where it was hanging on the WIFI and MQTT connected page!!!! Thank you so much!  I am back up and running.

**Reply by aderusha** [reply]
Thanks for the feedback!  I guess we're good, so moving this to release.

**Comment by the-real-techfreak** [comment]
This HASPone v1.02 works like a charm. Loads of thanks for this fix.

I just have one query actually. I have a fan at my place whose speed can be varied. I am trying to add it as a dimmer so that I can have Time, Dimmer Light and Fan in a single page (Page 4). But this dimmer page accepts only lights. Is there any turnaround for this to have it done. Currently I am struggling to do it by creating a dummy template light and linking it various speeds of the fan to bring it here.

In any case, kudos to your HASPone fix in a quick time.
